libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Generators;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    /// <summary>
+    /// An <see cref="IBot"/> implementation that manages the execution of an <see cref="AdaptiveDialog"/>.
+    /// </summary>
+    public class AdaptiveDialogBot : IBot
+    {
+        private static readonly ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager> _languageGeneratorManagers = new ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager>();
+
+        private readonly string _defaultLocale;
+        private readonly string _adaptiveDialogId;
+        private readonly string _languageGeneratorId;
+        private readonly ResourceExplorer _resourceExplorer;
+        private readonly SkillConversationIdFactoryBase _skillConversationIdFactoryBase;
+        private readonly ConversationState _conversationState;
+        private readonly UserState _userState;
+        private readonly BotFrameworkAuthentication _botFrameworkAuthentication;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AdaptiveDialogBot"/> class.
+        /// </summary>
+        /// <param name="adaptiveDialogId">The id of the <see cref="AdaptiveDialog"/> to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="languageGeneratorId">The id of the <see cref="LanguageGenerator"/> to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="defaultLocale">The default locale for this bot.</param>
+        /// <param name="resourceExplorer">The Bot Builder <see cref="ResourceExplorer"/> to load the <see cref="Dialog"/> from.</param>
+        /// <param name="conversationState">A <see cref="ConversationState"/> implementation.</param>
+        /// <param name="userState">A <see cref="UserState"/> implementation.</param>
+        /// <param name="skillConversationIdFactoryBase">A <see cref="SkillConversationIdFactoryBase"/> implementation.</param>
+        /// <param name="botFrameworkAuthentication">A <see cref="BotFrameworkAuthentication"/> used to obtain a client for making calls to Bot Builder Skills.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        public AdaptiveDialogBot(
+            string adaptiveDialogId,
+            string languageGeneratorId,
+            string defaultLocale,
+            ResourceExplorer resourceExplorer,
+            ConversationState conversationState,
+            UserState userState,
+            SkillConversationIdFactoryBase skillConversationIdFactoryBase,
+            BotFrameworkAuthentication botFrameworkAuthentication,
+            ILogger logger)
```

Comment:
Default it to null if we will be instantiating a NullLogger if not provided. 
```suggestion
            ILogger logger = null)
```